src/prefect/core/task.py

```diff
@@ -393,7 +396,7 @@ def __init__(
                         f"Invalid `retry_on` value {v!r}. Expected an exception subclass."
                     )
         else:
-            self.retry_on = set([])
+            self.retry_on = None
```

Comment:
Sorry to keep nitpicking, but you can set this to `None` on L382 and remove this branch